src/ray/core_worker/core_worker.cc

```diff
@@ -1441,57 +1404,45 @@ static std::vector<std::string> GetUrisFromRuntimeEnv(
   return result;
 }
 
-static std::vector<std::string> GetUrisFromSerializedRuntimeEnv(
-    const std::string &serialized_runtime_env) {
+std::string CoreWorker::OverrideTaskOrActorRuntimeEnv(
+    const std::string &serialized_runtime_env,
+    std::vector<std::string> *runtime_env_uris) {
+  std::shared_ptr<rpc::RuntimeEnv> parent_runtime_env;
```

Comment:
Why is it a shared pointer? Can we just make it `rpc::RuntimeEnv`? Also, this API  seems dangerous (worker_context_.GetCurrentRuntimeEnv()). The return value should be immutable, but now we can mutate it.

Reply:
Try to avoid object copy.
I can modify the return type to `std::shared_ptr<const rpc::RuntimeEnv>`, thanks.

Reply:
modified return type sounds good to me!